testsuite/python/mmm1d.py

```diff
@@ -20,33 +20,98 @@
 import unittest_decorators as utx
 import tests_common
 import espressomd
-
-if espressomd.has_features("ELECTROSTATICS", "PARTIAL_PERIODIC"):
-    from espressomd.electrostatics import MMM1D
+import numpy as np
 
 
-@utx.skipIfMissingFeatures(["ELECTROSTATICS", "PARTIAL_PERIODIC"])
-class ElectrostaticInteractionsTests(ut.TestCase):
+class ElectrostaticInteractionsTests(object):
     # Handle to espresso system
-    system = espressomd.System(box_l=[1.0, 1.0, 1.0])
+    system = espressomd.System(box_l=[10.0] * 3)
+    system.periodicity = [0, 0, 1]
+    system.time_step = 0.01
+    system.cell_system.skin = 0.4
+    system.cell_system.set_n_square()
+    system.thermostat.set_langevin(kT=0,gamma=1, seed=8)
 
+    pid_target, pos_x_target, pos_y_target, pos_z_target, q_target, f_x_target, f_y_target, f_z_target = np.loadtxt(
+        tests_common.abspath("data/mmm1d_data.txt"), unpack=True)
+    vec_f_target = np.stack((f_x_target, f_y_target, f_z_target), axis=-1)
+    energy_target = -7.156365298205383
+    num_particles = pid_target.shape[0]
+    pos_x_target=np.mod(pos_x_target,10)
```

Comment:
@fweik Hey Floh. What changed in the implementation of setting particles that setting them into the main box is needed to obtain the correct result?
Will the MMM1D calculation break if the particle leaves the main box and "enters" an image box? It seems like the implementation is sensitive to the periodic image... (This means the test fails if you comment out all lines with np.mod() )...

Reply:
the particles were set outside the non-periodic box. this resulted in wrong results. If Espresso would throw an error like suggested in #2905 this would have been obvious